picpro/ChipInfoEntry.py

```diff
@@ -120,6 +120,7 @@ def programming_vars(self) -> ProgrammingVars:
             program_retries=self.program_tries,
             over_program=self.over_program,
             fuse_blank=self.fuse_blank,
+            flash_chip=self.flash_chip,
```

Comment:
flash_chip does not need to be in ProgrammingVars since programming vars contain ~only data needed for ProtocolInterface.init_programming_vars, just use chip_info.flash_chip instead

Reply:
Got it. It looks like the code might need some refactoring to simplify the structure. Let me know if you’d like me to make the changes.